data/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -235,24 +235,30 @@ private FloatWriter(int id) {
     @Override
     public void nonNullWrite(int rowId, Float data, ColumnVector output) {
       ((DoubleColumnVector) output).vector[rowId] = data;
-      if (Float.isNaN(data)) {
-        nanCount++;
-      }
+      floatFieldMetricsBuilder.addValue(data);
+    }
+
+    @Override
+    public void nullWrite() {
+      nullValueCount++;
```

Comment:
It seems like it may be a good idea to move null tracking into the builder, so you could just run `metricsBuilder.addValue(null)` or `metricsBuilder.addNullValue()`. Then you wouldn't need to rebuild the metrics later.